Documentation: fix download URL for the community modules and extensions

jgrocha opened the pull request:
This PR replaces references to the old server ares.opengeo.org in the documentation. The protocol is also changed from http to https.

```bash
cd doc/en/user/source
find . -type f -name '*'.rst -exec sed -i 's/http:\/\/ares.opengeo/https:\/\/build.geoserver/g' {} +
```

Afterwards, I've also:
1. changed the pointers to the geofence (server and plugin) from community to extensions,
1. fix some URLs to make them all terminating with a slash.

nmco commented on doc/en/user/source/extensions/mongodb/index.rst:
```diff
@@ -1,40 +1,40 @@
-   *Configuring a MongoDB data store.*
-
-For more details about the usage of this data store please check the `GeoTools MongoDB plugin documentation
+.. _mongodb:
```
Not sure why this file is all marked as changed, something obvious I'm missing ? That say, it looks good.

jgrocha replied:
Let me check if it is the annoying LF/CRLF problem.

nmco commented on doc/en/user/source/extensions/geofence-server/installing.rst:
```diff
@@ -3,7 +3,7 @@
 ==================================================
 
- #. Download the extension from the `nightly GeoServer community module builds <http://ares.opengeo.org/geoserver/master/community-latest/>`_.
+ #. Download the extension from the `nightly GeoServer community module builds <https://build.geoserver.org/geoserver/master/ext-latest/>`_.
```
Well since GeoFence is an extension now, the text should be updated too, usually for extensions something like this is used:
```

#. Navigate to the `GeoServer download page <http://geoserver.org/download>`_.
#. Find the page that matches the version of the running GeoServer.
#. Download the GeoFence extension. The download link will be in the :guilabel:`Extensions` 
   section under :guilabel:`Other`.
#. Extract the files in this archive to the :file:`WEB-INF/lib` directory of your GeoServer installation.
#. Restart GeoServer
```

Although since GeoFence is only available in master as an extension I'm not sure where it will appear on the download page,  @afabiani any idea ?

jgrocha replied:
@nmco It does not appear on previous versions. We need to point to `master/ext-latest/` right now.

nmco commented on doc/en/user/source/extensions/geofence/installing.rst:
```diff
@@ -3,7 +3,7 @@
 Installing the GeoServer GeoFence extension
 ===========================================
 
- #. Download the extension from the `nightly GeoServer community module builds <http://ares.opengeo.org/geoserver/master/community-latest/>`_.
```
Same comment as above.

nmco commented:
Just some minor comment, thanks for the contribution @jgrocha !

jgrocha commented:
@nmco Thank you for your comments. I've changed the MongoDB intslattion instructions to follow the same pattern of other extensions.

I've also added a note to Geofence installation (client and server extensions) to highlight the change from community to extension.
